runtime/model-protobuf/src/main/java/io/aklivity/zilla/runtime/model/protobuf/internal/ProtobufParser.java

```diff
@@ -120,26 +135,32 @@ private FileDescriptor parseProto3(CharStream input)
         }
         catch (DescriptorValidationException ex)
         {
-            throw new ProtobufParseException("Failed to parse Proto3 schema", ex);
-        }
-        catch (Exception ex)
-        {
-            throw new ProtobufParseException("Unexpected error parsing Proto3 schema", ex);
+            LangUtil.rethrowUnchecked(ex);
         }
+        return descriptor;
     }
 
-    public static class ProtobufParseException extends RuntimeException
+    public static ProtoSyntax detectSyntax(String schemaText)
     {
-        private static final long serialVersionUID = 1L;
-
-        public ProtobufParseException(String message)
+        if (schemaText == null || schemaText.isEmpty())
         {
-            super(message);
+            return ProtoSyntax.UNKNOWN;
         }
 
-        public ProtobufParseException(String message, Throwable cause)
+        Matcher matcher = SYNTAX_PATTERN.matcher(schemaText);
+        if (matcher.find())
         {
-            super(message, cause);
+            String syntax = matcher.group(1).toLowerCase();
+            if (PROTO2_SYNTAX.equals(syntax))
+            {
+                return ProtoSyntax.PROTO2;
+            }
+            else if (PROTO3_SYNTAX.equals(syntax))
+            {
+                return ProtoSyntax.PROTO3;
+            }
         }
+        return ProtoSyntax.UNKNOWN;
```

Comment:
```suggestion
        return matcher.find() ? ProtoSyntax.of(matcher.group(1).toLowerCase()) : ProtoSyntax.UNKNOWN;
```

This would be more readable if you used a named group called `syntax`, instead of a numbered group, in the pattern.